Remotion/Data/DomainObjects/DomainObject.cs

```diff
@@ -463,7 +468,7 @@ public override string ToString ()
     /// <remarks>To perform custom actions when a <see cref="DomainObject"/> is deleted <see cref="OnDeleting"/> and <see cref="OnDeleted"/> should be overridden.</remarks>
     protected void Delete ()
     {
-      CheckInitializeEventNotExecuting();
+      _strategy?.ThrowIfReferenceInitializeEventExecuting();
```

Comment:
we should move `_strategy.ThrowIfReferenceInitializeEventExecuting()` into `CheckInitializeEventNotExecuting`. This way, we keep the semantic simple and in case further requirements are added in the future, have an existing helper method to fall back on. Law of Demeter can also apply inside a class.